spring-integration-core/src/main/java/org/springframework/integration/gateway/GatewayProxyFactoryBean.java

```diff
@@ -279,6 +280,16 @@ protected AsyncTaskExecutor getAsyncExecutor() {
 		return this.asyncExecutor;
 	}
 
+	/**
+	 * Return the Map of {@link Method} to {@link MessagingGatewaySupport}
+	 * generated by this factory bean.
+	 * @return the map.
+	 * @since 4.3
+	 */
+	public Map<Method, MessagingGatewaySupport> getGateways() {
+		return Collections.unmodifiableMap(this.gatewayMap);
```

Comment:
Should be this:
```
return Collections.<Method, MessagingGatewaySupport> unmodifiableMap(this.gatewayMap);
```
We may reconsider when will come to Java 8 finally :smile: 
Will fix on merge.